selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java

```diff
@@ -72,20 +86,81 @@ public void setUp() throws Exception {
+    consoles.waitExpectedTextIntoConsole(FINISH_LANGUAGE_SERVER_INITIALIZATION_MESSAGE);
   }
 
   @Test
+  public void checkGoToSymbolFeature() {
+    editor.goToCursorPositionVisible(14, 7);
+    List<String > expectedListSymbols =
+            Arrays.asList(
+                    "greeter",
+                    "Greeter",
+                    "greet",
+                    "constructor",
+                    "\"Greeter\"",
+                    "print",
+                    "printVar",
+                    "greeting");
+    menu.runCommand(ASSISTANT, GO_TO_SYMBOL);
+    assistantFindPanel.waitAllNodes(expectedListSymbols);
+    assistantFindPanel.clickOnActionNodeWithTextEqualsTo("greet");
+    editor.waitCursorPosition(19, 5);
+  }
+
+  @Test(priority = 1, alwaysRun = true)
+  public void checkGoToSymbolInWorkspaceFeature() {
+    projectExplorer.waitAndSelectItem(PROJECT_NAME);
+    menu.runCommand(ASSISTANT, FIND_PROJECT_SYMBOL);
+    assistantFindPanel.typeToInputField("testPrint");
+    assistantFindPanel.clickOnActionNodeWithText("testPrint");
+    editor.waitCursorPosition(26, 6);
+  }
+
+  @Test(priority = 2, alwaysRun = true)
   public void checkMainFeaturesTypeScriptLS() {
     String intitTypeScriptLanguageServerMessage =
-        String.format(
-            "Finished language servers initialization, file path '/%s'", PATH_TO_GREETER_FILE);
+        format("Finished language servers initialization, file path '/%s'", PATH_TO_GREETER_FILE);
 
     consoles.waitExpectedTextIntoConsole(intitTypeScriptLanguageServerMessage);
     checkCodeValidation();
     checkCodeAssistant();
     checkGoToDefinition();
   }
 
+  @Test(priority = 3, alwaysRun = true)
+  public void checkFindReferencesFeature() {
+    String referenceInGreeterClass =
+        format("/%s/Greeter.ts\n" + "From:24:17 To:24:22", PROJECT_NAME);
+    String referenceInTestPrintClass =
+        format("/%s/testPrint.ts\n" + "From:14:0 To:14:5", PROJECT_NAME);
+    menu.runCommand(ASSISTANT, FIND_REFERENCES);
+    findReferencesConsoleTab.waitAllReferencesWithText(
+        referenceInGreeterClass, referenceInTestPrintClass);
+    findReferencesConsoleTab.doubleClickOnReference(referenceInGreeterClass);
+    editor.waitCursorPosition(25, 23);
+  }
+
+  @Test(priority = 4, alwaysRun = true)
+  public void checkHoveringFeature() {
+    editor.moveCursorToText("Greeter");
+    editor.waitTextInHoverPopUpEqualsTo(
+        "class Greeter", "Known permanent failure: ihttps://github.com/eclipse/che/issues/10699");
+  }
+
+  @Test(priority = 5, alwaysRun = true)
+  public void checksignatureHelpProvider() {
+    editor.goToCursorPositionVisible(25, 38);
+    editor.typeTextIntoEditor(ENTER.toString());
+    editor.typeTextIntoEditor("printVar.print(");
+
+    try {
+      editor.waitExpTextIntoShowHintsPopUp("");
+    } catch (WebDriverException ex) {
+      fail("Known permanent failure: issue https://github.com/eclipse/che/issues/11324");
```

Comment:
**WebDriverException** is too general here. So, it would be useful to have exact error in logs:
```
fail("Known permanent failure: issue https://github.com/eclipse/che/issues/11324", ex);
```